src/js_network_transport.hpp

```diff
@@ -179,7 +179,7 @@ struct JavaScriptNetworkTransport : public app::GenericNetworkTransport {
     }
 
     using SendRequestHandler = void(ContextType m_ctx, const app::Request request, std::function<void(const app::Response)> completion_callback);
-    using NetworkTransportFactory = std::function<std::unique_ptr<app::GenericNetworkTransport>(ContextType, Dispatcher)>;
+    using NetworkTransportFactory = std::unique_ptr<app::GenericNetworkTransport>(*)(ContextType, Dispatcher);
```

Comment:
We should probably leave a comment here as to why the C-style function pointer is necessary.  Just so someone doesn't "clean up" the code and re-introduce the bug.

Reply:
:+1: